Yatima/Expr.lean

```diff
@@ -31,6 +31,7 @@ inductive Expr
   | lit   : Literal → Expr
   | lty   : LitType → Expr
   | fix   : Name → Expr → Expr
+  | proj  : Nat → Expr → Expr
```

Comment:
doesn't this need to be `proj : Nat -> ConstCid -> Expr -> Expr`? My understanding was that we needed the `Inductive` constant that we're projecting, or is the intent that the `Expr` here is an `Expr.const` reference?

Reply:
If we did have the type of the inductive in `proj` though, wouldn't we then be able to just check that the type of the `Expr` is the `inductive`, thus converting an `infer` to a `check` and saving some work?

My point is just that it seems like we already have this info when we compile from Lean, and it doesn't impact content-addressing to include it. If you think it's neutral or detrimental to do so, then let's leave it out, but my instinct is that it's a small benefit to have it